internal/fs/fs.go

```diff
@@ -1629,6 +1639,16 @@ func (fs *fileSystem) RmDir(
 	//     https://github.com/GoogleCloudPlatform/gcsfuse/issues/9
 	//
 	//
+
+	// Check for local file entries.
+	localFileEntries := childDir.LocalFileEntries(fs.localFileInodes)
+	// Are there any local entries?
+	if len(localFileEntries) != 0 {
```

Comment:
This is just a validation right. when does localFileEntries be present in dir if unlink on files is called first?

Reply:
When `os.RemoveAll` is called, RmDir call is made first followed by unlink calls.
Whereas in case of `rm -r` unlink calls are made first. 
Ref: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1311/files/3053d4a168dd5ccb7bcaceba933fd4451f21cd8e#diff-ef03141dd8338143c09a5b30ceb3d079d25015a96f8b5292ee29daa23348df2eR1590